homeassistant/components/sensor/steam_online.py

```diff
@@ -0,0 +1,88 @@
+"""
+homeassistant.components.sensor.steam_online
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Sensor for Steam account status.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.steam_online/
+"""
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import (ATTR_ENTITY_PICTURE, CONF_API_KEY)
+
+STATE_OFFLINE = 'Offline'
+STATE_ONLINE = 'Online'
+STATE_BUSY = 'Busy'
+STATE_AWAY = 'Away'
+STATE_SNOOZE = 'Snooze'
+STATE_TRADE = 'Trade'
+STATE_PLAY = 'Play'
+ICON = 'mdi:steam'
+
+REQUIREMENTS = ['steamodd==4.21']
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Steam platform. """
+    api_key = config.get(CONF_API_KEY)
+    add_devices(
+        [SteamSensor(account, api_key) for account in config.get('accounts', [])])
+
+class SteamSensor(Entity):
+    """ Steam account. """
+
+    # pylint: disable=abstract-method
+    def __init__(self, account, api_key):
+        self._api_key = api_key
+        self._account = account
+        self._state = STATE_OFFLINE
+        self.update()
+
+    @property
+    def should_poll(self):
+        """ Device should be polled. """
+        return True
+
+    @property
+    def name(self):
+        """ Returns the name of the sensor. """
+        return str(self._account)
+
+    @property
+    def state(self):
+        """ State of the sensor. """
+        return self._state
+
+    # pylint: disable=no-member
+    def update(self):
+        """ Update device state. """
+        import steam
```

Comment:
I would move some of the initializations around.
 - import steam and set api key in `setup_platform` so it only happens once
 - pass steam object into SteamSensor
 - limit `update()` to just `self._profile = steam.user.profile(self._account)`
 - Move which state to return to `state` property
 - Move avatar_medium to be in `device_state_attributes`